### ISSUE TITLE
Open Source the Backend for Self Hosting

### ISSUE THREAD

**Issue by tnyeanderson** [klinker-apps/messenger-issues#938]
- [x] I have [searched](https://github.com/klinker-apps/messenger-issues/issues), by keyword, for my request to verify that there aren't any open or closed issues similar to this behavior.
- [x] I have [searched the help page](https://messenger.klinkerapps.com/help) to verify that there are not any topics discussing similar behavior.

**Describe your request:**
I love Pulse. I've been using it for a couple years now and it never ceases to amaze me. I find it much more stable than any competitor, even iMessage. And the fact that you offer it at the price you do is seriously mind-blowing and a complete gamechanger in the market. There is no competitor to Pulse!

But I also love self hosting and being in full control of my data. I understand that this is almost definitely not an option (and with good reason, à la #75) but it would be the most incredible thing to open source the backend (perhaps a docker solution?) so I could spin up a Pulse server, plug my new URL into the Android app (which would still be closed source) and have my messages be stored on and transferred to and from my own equipment. This would obviously add some complications, but it would give the privacy-minded/selfhosted community an option that is otherwise COMPLETELY unavailable in the marketplace. I think the vast majority of users wouldn't even try to mess with it anyway, so it may not have an appreciable affect on your userbase... but I can't speak to the effect it would have on your IP, revenue, etc. You know that better than I would.

Obviously Pulse is a valuable asset for you, and I completely understand why you wouldn't want to open source the backend... but I figured I'd throw it into the aether anyway. Sincerely, thank you for the hard work on this invaluable app!

**Comment by klinker24** [comment]
Sorry, that just isn't planned at this time. Everything stored in Pulse is end to end encrypted, so privacy really isn't a concern

https://messenger.klinkerapps.com/encryption.html

I feel as though my contributions through Pulse and Android in general are numerous and exemplary of my appreciation of the open source community. Open sourcing the backend would mean that I would be open sourcing the Android app as well. I would be letting developers take the full-stack into their own hands, I wouldn't just give an option in the Android app to point to a different backend.

I obviously love open source work and contribute to open source projects (my own and others) very regularly. I do realize that there are plenty of reasons - other than privacy - that people would want the backend open sourced. However, my opinions on it have not really changed since discussed in #75. At this time, that is just not something that I can do.

**Comment by tnyeanderson** [comment]
I completely understand, and that was the answer I expected. THANK YOU for your contributions to open source, THANK YOU for responding so quickly to all of your users, and THANK YOU for creating the best SMS app ever. I'm not going anywhere!

**Comment by tnyeanderson** [comment]
And just one year later, @klinker24 betrayed the trust of myself and his entire user base. Looks like it's time to look elsewhere.

You ruined this, Luke. You really did. But hey, at least you made a lot of money right? How great for you; how pitiful for your users.

**Comment by klinker24** [comment]
Interesting that you claim to speak for the entire user base. You don't. I have received nothing but outpourings of gratitude for creating such an amazing product and putting the time and work into it that I did. It seems that everyone else contacting me couldn't be more thrilled that I was able to walk away with a great payoff for the years and years of work I put in, and are excited to see what the new owner of the app brings to the table.

I betrayed no one and you haven't even given the new owner of the app a chance. They can put more time, resources, and energy into growing the app than I ever could. If you had actually read the email that they sent to all of the app's users yesterday, you also would have seen that they are going to be following the exact same principles of privacy, security, and encryption that I always have. Absolutely nothing has or will be changing around that. You have nothing to complain about nor any reason to rant at me.

I would ask that you never come to me or any other developer with that attitude again. It is not appreciated, nor will it get you anywhere. I don't own the app anymore, so there is no reason for you to ever contact me again. Thanks and have a great day.
